packages/cli/src/controllers/oauth/abstract-oauth.controller.ts

```diff
@@ -125,12 +125,16 @@ export abstract class AbstractOAuthController {
 		credential: ICredentialsDb,
 		decryptedData: ICredentialDataDecryptedObject,
 		additionalData: IWorkflowExecuteAdditionalData,
+		canUseSecrets: boolean = false,
 	) {
 		return this.credentialsHelper.applyDefaultsAndOverwrites(
 			additionalData,
 			decryptedData,
 			credential.type,
 			'internal',
+			undefined,
+			undefined,
+			canUseSecrets,
```

Comment:
perhaps instead of passing this around in so many places, we should update `credentialsHelper.applyDefaultsAndOverwrites` to directly use `this.credentialCanUseExternalSecrets`, and make `credentialCanUseExternalSecrets` private.
That way we stop leaking this abstraction.